File: src/main/java/io/vertigo/shiny/models/text/toggle/ShinyToggle.java
package io.vertigo.shiny.models.text.toggle;

import io.vertigo.shiny.models.ShinyModel;

public record ShinyToggle(
		String label,
		boolean value,
		ShinyToggleType toggleType,
		String onText,
		String offText,
		boolean showText) implements ShinyModel {

	public String shinyType() {
		return "ShinyToggle";
	}
}

File: banshee/src/models/input/toggle/toggle/toggle/ShinyToggle.ts
import { ShinyModel } from '../../../ShinyModel';
import { ShinyToggleType } from './ShinyToggleType';

export interface ShinyToggle extends ShinyModel {
  label: string;
  value: boolean;
  toggleType: ShinyToggleType;
  onText?: string;
  offText?: string;
  showText: boolean;
}
